far/dirinfo.cpp

```diff
@@ -111,40 +111,44 @@ int GetDirInfo(string_view const DirName, DirInfoData& Data, multifilter* Filter
 	// Временные хранилища имён каталогов
 	string strLastDirName;
 	os::fs::find_data FindData;
+
+	// Mantis#0002692
+	struct SuspendMacros
+	{
+		SuspendMacros() { Global->CtrlObject->Macro.SuspendMacros(true); }
+		~SuspendMacros() { Global->CtrlObject->Macro.SuspendMacros(false); }
+	} Suspend;
+
 	while (ScTree.GetNextName(FindData,strFullName))
 	{
-		// Mantis#0002692
-		if (!Global->CtrlObject->Macro.IsExecuting())
-		{
-			INPUT_RECORD rec;
+		INPUT_RECORD rec;
 
-			switch (PeekInputRecord(&rec))
-			{
-				case NO_KEY:
-					break;
-				case KEY_NONE:
-				case KEY_ALT:
-				case KEY_CTRL:
-				case KEY_SHIFT:
-				case KEY_RALT:
-				case KEY_RCTRL:
-				case KEY_LWIN: case KEY_RWIN:
-					GetInputRecordNoMacroArea(&rec);
-					break;
-				case KEY_ESC:
-				case KEY_BREAK:
-					GetInputRecordNoMacroArea(&rec);
-					return 0;
-				default:
-
-					if (Flags&GETDIRINFO_ENHBREAK)
-					{
-						return -1;
-					}
+		switch (PeekInputRecord(&rec))
+		{
+			case NO_KEY:
+				break;
+
+			case KEY_ESC:
+			case KEY_BREAK:
+				GetInputRecordNoMacroArea(&rec);
+				return 0;
+
+			case KEY_LEFT:
+			case KEY_RIGHT:
+			case KEY_DOWN:
+			case KEY_UP:
+			case KEY_PGDN:
+			case KEY_PGUP:
+			case KEY_HOME:
+			case KEY_END:
```

Comment:
The idea of "enhbreak" is to cancel the scan when you move around the file panel with quick view open.

And there are way more ways to move around the file panel than these. For example:
- numpad keys
- mouse clicks (all 3)
- mouse wheel
- Alt+letter
- Folder shortcuts
- ...

On top of that, currently any command line input (i.e. any printable key) also cancels the scan.

Reply:
Reverted